receiver/awscontainerinsightreceiver/internal/stores/podstore.go

```diff
@@ -135,23 +135,36 @@ func NewPodStore(hostIP string, prefFullPodName bool, addFullPodNameMetricLabel
 
 	nodeName := os.Getenv(ci.HostName)
 	if nodeName == "" {
-		return nil, fmt.Errorf("missing environment variable %s. Please check your deployment YAML config", ci.HostName)
+		nodeName = hostName
+		if nodeName == "" {
+			return nil, fmt.Errorf("missing environment variable %s. Please check your deployment YAML config or passed as part of the agent config", ci.HostName)
+		}
+	}
+
+	k8sClient := &k8sclient.K8sClient{}
+	nodeInfo := &nodeInfo{
+		nodeName: nodeName,
+		provider: nil,
```

Comment:
Isn't this going to cause panics? None of the `nodeInfo` methods check if the provider is nil before calling its methods.

Reply:
thats because thats protected by checking if the k8s client is not nill